service/definition/service.go

```diff
@@ -14,20 +18,20 @@ type ServiceDefinition struct {
 	// Description is service description.
 	Description string `yaml:"description" json:"description,omitempty" validate:"printascii"`
 
+	// Repository holds the service's repository url if it's living on a git host.
+	Repository string `yaml:"repository" json:"repository,omitempty" validate:"omitempty,uri"`
+
 	// Tasks are the list of tasks that service can execute.
 	Tasks map[string]*Task `yaml:"tasks" json:"tasks,omitempty" validate:"dive,keys,printascii,endkeys,required"`
 
 	// Events are the list of events that service can emit.
 	Events map[string]*Event `yaml:"events" json:"events,omitempty" validate:"dive,keys,printascii,endkeys,required"`
 
-	// Dependencies are the Docker containers that service can depend on.
-	Dependencies map[string]*Dependency `yaml:"dependencies" json:"dependencies,omitempty" validate:"dive,keys,printascii,ne=service,endkeys,required"`
-
 	// Configuration is the Docker container that service runs inside.
-	Configuration *Dependency `yaml:"configuration" json:"configuration,omitempty"`
+	Configuration Dependency `yaml:"configuration" json:"configuration,omitempty"`
```

Comment:
Is there a reason to remove the pointer here but keep them on Tasks, Events, etc..?

Could we remove all pointer?